onchange_helper/models/models.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+# © 2016 Akretion (http://www.akretion.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import api, models
+
+
+def get_new_values(model, record, on_change_result):
+    vals = on_change_result.get('value', {})
+    new_values = {}
+    for fieldname, value in vals.iteritems():
+        if fieldname not in record:
+            column = model._fields[fieldname]
+            if column.type == 'many2one':
+                value = value[0]  # many2one are tuple (id, name)
+            new_values[fieldname] = value
+    return new_values
+
+
+@api.model
+def play_onchanges(self, values, onchange_fields):
+    onchange_specs = self._onchange_spec()
+
+    # we need all fields in the dict even the empty ones
+    # otherwise 'onchange()' will not apply changes to them
+    all_values = values.copy()
+    for field in self._fields:
+        if field not in all_values:
+            all_values[field] = False
+
+    # we work on a temporary record
+    new_record = self.new(all_values)
+
+    new_values = {}
+    for field in onchange_fields:
+        onchange_values = new_record.onchange(all_values,
+                                              field, onchange_specs)
+        new_values.update(get_new_values(self, values, onchange_values))
+        all_values.update(new_values)
+
+    res = {f: v for f, v in all_values.iteritems()
+           if f in values or f in new_values}
+    return res
+
+models.Model.play_onchanges = play_onchanges
```

Comment:
@florian-dacosta Take a look at https://github.com/OCA/server-tools/blob/9.0/base_suspend_security/models/ir_rule.py#L33 to see how you can safely add new method to base model in odoo < 10